tests/testthat/test-summarisePopulationCharacteristics.R

```diff
@@ -0,0 +1,117 @@
+                    c("overall")))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Number records") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    2694))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Cohort end date" & estimate_name == "max") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    "2010-01-01"))
+
+  # Add sex and age group strata
+  expect_no_error(summarisedPopulation <- summarisePopulationCharacteristics(
+    cdm = cdm,
+    studyPeriod = c("1950-01-01", NA),
+    sex = TRUE,
+    ageGroup = list(c(0,20),c(21,150)))
+    )
+  expect_true(inherits(summarisedPopulation,"summarised_result"))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::select("strata_name") |>
+                    dplyr::distinct() |>
+                    dplyr::pull() ==
+                    c("overall", "sex", "age_group", "sex &&& age_group")))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Number records") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    c(2693,1372,1321,2521,172,1271,101,1250,71)))
+  expect_true(summarisedPopulation |>
+                dplyr::filter(variable_name == "Age at end" & strata_level == "0 to 20" & estimate_name == "min") |>
+                dplyr::pull("estimate_value") <
+                summarisedPopulation |>
+                dplyr::filter(variable_name == "Age at end" & strata_level == "21 to 150" & estimate_name == "min") |>
+                dplyr::pull("estimate_value"))
+
+  # Only sex
+  expect_no_error(summarisedPopulation <- summarisePopulationCharacteristics(
+    cdm = cdm,
+    sex = TRUE
+  ))
+  expect_true(inherits(summarisedPopulation,"summarised_result"))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::select("strata_name") |>
+                    dplyr::distinct() |>
+                    dplyr::pull() ==
+                    c("overall", "sex")))
+
+  # Only age group
+  expect_no_error(summarisedPopulation <- summarisePopulationCharacteristics(
+    cdm = cdm,
+    ageGroup = list(c(0,1), c(2,Inf))
+  ))
+  expect_true(inherits(summarisedPopulation,"summarised_result"))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::select("strata_name") |>
+                    dplyr::distinct() |>
+                    dplyr::pull() ==
+                    c("overall", "age_group")))
+
+  PatientProfiles::mockDisconnect(cdm = cdm)
+})
+
+
+test_that("summarisePopulationCharacteristics() expected errors", {
+  # Load mock database ----
+  cdm <- cdmEunomia()
+
+  expect_error(summarisePopulationCharacteristics("cdm"))
+  expect_error(summarisePopulationCharacteristics(cdm, studyPeriod = NULL))
```

Comment:
should studyPeriod = NULL work? It could be equivalent to c(NA, NA). To consider

Reply:
decided that this should work